### PR TITLE
CA Removal

### PR DESCRIPTION
**Pull request by Magisus**
* (PUP-8998) Remove all CA and certificate-related faces
* (PUP-9022) Disable the `puppet cert` command
* (PUP-8912) Remove the CertificateAuthority

### CODE REVIEW

**Comment by Magisus** [comment]
I think the Travis failure will be fixed by https://github.com/puppetlabs/puppet/pull/6989, which updates the Rakefile to not require packaging for non-packaging tasks.

**Comment by Magisus** [comment]
There are a number of acceptance tests that have to be removed as part of this, but they should be reworked and added to puppetserver's tests instead.

**Comment by Magisus** [comment]
jenkins please test this

**Comment by Magisus** [comment]
jenkins please test this

**Comment by Magisus** [comment]
jenkins please test this

**Comment by Magisus** [comment]
Where did Travis go? Closing and reopening.

**Comment by Magisus** [comment]
Still no Travis... but apparently PR testing? (expected to fail)

File: lib/puppet/face/node/clean.rb [hunk 1]
```diff
@@ -1,19 +1,18 @@
 Puppet::Face.define(:node, '0.0.1') do
   action(:clean) do
 
-    summary _("Clean up signed certs, cached facts, node objects, and reports for a node stored by the puppetmaster")
```
Comment (justinstoller) [inline lib/puppet/face/node/clean.rb]: This is wrong now. We updated this face to attempt to load the puppetserver-ca gem, see below.

File: lib/puppet/application/cert.rb [hunk 2]
```diff
@@ -100,219 +99,19 @@ def subcommand=(name)
...
-------
-Luke Kanies
-
-
-COPYRIGHT
----------
-Copyright (c) 2011 Puppet Inc., LLC Licensed under the Apache 2.0 License
-
-    HELP
+    _("This command is no longer functional, please use `puppetserver ca` instead.")
```
Comment (justinstoller) [inline lib/puppet/application/cert.rb]: It'd be nice to update this to exactly what folks can use instead.

File: lib/puppet/ssl/certificate_factory.rb [hunk 3]
```diff
@@ -19,12 +19,11 @@ module Puppet::SSL::CertificateFactory
   # @param serial [Integer] The serial number for the given certificate, which
   #   MUST be unique for the given CA.
   # @param ttl [String] The duration of the validity for the given certificate.
-  #   defaults to Puppet[:ca_ttl]
   #
   # @api public
   #
   # @return [OpenSSL::X509::Certificate]
-  def self.build(cert_type, csr, issuer, serial, ttl = nil)
+  def self.build(cert_type, csr, issuer, serial, ttl = 3600)
```
Comment (justinstoller) [inline lib/puppet/ssl/certificate_factory.rb]: I assume the ttl change here was because we were deprecating `ca_ttl`? Since we're not we can probably undo this change. If we wanted to keep it we'd at least need to change the default (ca_ttl defaults to 5y, not 1hr).

**Comment by justinstoller** [comment]
I think we can also remove CertificateFactory and its tests.

**Comment by justinstoller** [comment]
We can also get rid of the settings `ca`, `capass`, and `caprivatedir` as part of this.

File: lib/puppet/application/cert.rb (hunk 2 above)
Comment (Magisus) [reply lib/puppet/application/cert.rb]: Like per command?

**Comment by Magisus** [comment]
Updated.

File: lib/puppet/ssl/certificate_factory.rb (hunk 3 above)
Comment (justinstoller) [reply lib/puppet/ssl/certificate_factory.rb]: Arggh! You changed the file location, and changed content, kept the git history, and the diff is even displayed properly in the web view!!!

You're going to need to show me how you did this, I must be missing some flag in my `git mv`ing repertoire.

Comment (Magisus) [reply lib/puppet/ssl/certificate_factory.rb]: Uhhh... I did literally nothing. Used shell's built-in `mv` :P

**Comment by justinstoller** [comment]
One of my test fix commits got rebased away? You need to remove `spec/integration/ssl/certificate_authority_spec.rb`.
